### ISSUE TITLE
Dependencies not showing on GitHub Enterprise Cloud repo

### ISSUE THREAD

**Issue by nigelcharman** [mikepenz/gradle-dependency-submission#96]
I'm not seeing any dependencies show up on the Dependency Graph. This is a private repository in GitHub Enterprise Cloud. 

![image](https://user-images.githubusercontent.com/144202/187288558-39d5cb99-ee82-4aac-a9ce-cc2bd2e57b84.png)

Our GitHub actions steps include:

```
    steps:
      - name: Checkout code
        uses: actions/checkout@v3
        
      - name: Calculates Gradle dependencies and submit to GitHubs Dependency Submission API
        # Currently the action has to be run on the default branch.
        # Running it as part of a PR won't update the Dependency graph of your projects Insights.
        if: ${{ github.ref == 'refs/heads/main' }}
        uses: mikepenz/gradle-dependency-submission@v0.7.0
        with:
          gradle-build-configuration: "testCompileClasspath"
```

The workflow run is showing:

![image](https://user-images.githubusercontent.com/144202/187288891-90d98234-092c-42c4-bc74-578de1801c14.png)

Running `gradlew dependencies` shows lots of dependencies for `testCompileClasspath`.

We have the Dependency Graph enabled:

![image](https://user-images.githubusercontent.com/144202/187287046-1ada272b-ccbe-4e5b-bd60-fb2ead9e25d6.png)

Any thoughts on what we're doing wrong?

**Comment by mikepenz** [comment]
It's worth noting that this requirement is only in place for the `--property` filter on the `properties` task. (to only retrieve particular properties) 

We do this so we don't have to write a properties output parser to filter out unwanted properties -> and more importantly the action should not request any information it is not meant to see -> request as little as needed.

In the action (highlighting for self-audit reasons) we use the `properties` task to retrieve the project `name` and `buildFile` path for modules:
https://github.com/mikepenz/gradle-dependency-submission/blob/main/src/gradle.ts#L91-L107

**Comment by nigelcharman** [comment]
I'm still not seeing anything. Do I need to wait a certain amount of time for them to show up?

The debug log is:

```
##[debug]Evaluating condition for step: 'Calculates Gradle dependencies and submit to GitHubs Dependency Submission API'
##[debug]Evaluating: (success() && (github.ref == 'refs/heads/main'))
##[debug]Evaluating And:
##[debug]..Evaluating success:
##[debug]..=> true
##[debug]..Evaluating Equal:
##[debug]....Evaluating Index:
##[debug]......Evaluating github:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'ref'
##[debug]....=> 'refs/heads/main'
##[debug]....Evaluating String:
##[debug]....=> 'refs/heads/main'
##[debug]..=> true
##[debug]=> true
##[debug]Expanded: (true && ('refs/heads/main' == 'refs/heads/main'))
##[debug]Result: true
##[debug]Starting: Calculates Gradle dependencies and submit to GitHubs Dependency Submission API
##[debug]Loading inputs
##[debug]Evaluating: github.token
##[debug]Evaluating Index:
##[debug]..Evaluating github:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'token'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Loading env
Run mikepenz/gradle-dependency-submission@v0.7.0
::group::📘 Reading input values
📘 Reading input values

::set-output name=snapshot::{"detector":{"name":"mikepenz/gradle-dependency-submission","url":"https://github.com/mikepenz/gradle-dependency-submission","version":"0.7.0"},"version":0,"job":{"correlator":"run_features--testCompileClasspath","id":"[2](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:2)951816476"},"sha":"0c0c9ae11a54457fdfbb6c2[3](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:3)9c68d[4](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:4)d160096112","ref":"refs/heads/main","scanned":"2022-08-29T21:1[5](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:5):52.734Z","manifests":{}}
##[debug]='{"detector":{"name":"mikepenz/gradle-dependency-submission","url":"https://github.com/mikepenz/gradle-dependency-submission","version":"0.7.0"},"version":0,"job":{"correlator":"run_features--testCompileClasspath","id":"295181[6](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:6)4[7](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:7)6"},"sha":"0c0c9ae11a54457fdfbb6c239c6[8](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:8)d4d1600[9](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:9)6[11](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:11)2","ref":"refs/heads/main","scanned":"2022-08-29T21:15:52.734Z","manifests":{}}'
Notice: Submitting snapshot...
Notice: {
    "detector": {
        "name": "mikepenz/gradle-dependency-submission",
        "url": "https://github.com/mikepenz/gradle-dependency-submission",
        "version": "0.7.0"
    },
    "version": 0,
    "job": {
        "correlator": "run_features--testCompileClasspath",
        "id": "2951816476"
    },
    "sha": "0c0c9ae11a54457fdfbb6c239c68d4d1600961[12](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:12)",
    "ref": "refs/heads/main",
    "scanned": "2022-08-29T21:[15](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:15):52.734Z",
    "manifests": {}
}
::endgroup::

Notice: Snapshot successfully created at [20](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:20)22-08-29T[21](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:21):15:53.[46](https://github.com/myorg/eoss-integration-features/runs/8080097051?check_suite_focus=true#step:3:48)3Z
##[debug]Node Action run completed with exit code 0
##[debug]Finishing: Calculates Gradle dependencies and submit to GitHubs Dependency Submission API
```

**Comment by mikepenz** [comment]
Oh I am sorry, I should have seen this directly. You are not configuring any `module`. 

As such this will be length of 0: https://github.com/mikepenz/gradle-dependency-submission/blob/main/src/main.ts#L22
Ultimately not doing any work: https://github.com/mikepenz/gradle-dependency-submission/blob/main/src/main.ts#L61-L72

If you want the `root` module, please use `:` as module spec. 

The README for example shows it like this:

```
- name: Root Gradle Dependency Submission
  uses: mikepenz/gradle-dependency-submission@{latest}
  with:
    use-gradlew: false
    gradle-build-module: |-
      :
    gradle-build-configuration: |-
      compileClasspath
```

Definitely need to add back to fail the build for this case.

**Comment by nigelcharman** [comment]
Thanks, that's working now @mikepenz.

Maybe it could default to `:` if no module is specified?

Next, I have to figure out how to get our internal repositories to not show up as FAILED :)

![image](https://user-images.githubusercontent.com/144202/187311852-4f04666d-f174-4e59-823c-34396b5d1aa0.png)

**Comment by mikepenz** [comment]
@nigelcharman for the `FAILED` one. how does the dependencies output look like for that one?

**Comment by nigelcharman** [comment]
I've just realised the step doesn't have the secrets to access the internal repositories. I've added those now and will let you know if it's fixed once the PR is merged.

**Comment by mikepenz** [comment]
Released v0.7.1 which has a log and falls back to `:` if not provided. thanks again for the report.

**Comment by nigelcharman** [comment]
I can confirm that the internal dependencies which previously showed as FAILED are picked up correctly with the credentials supplied to the workflow
